zstd.go

```diff
@@ -8,23 +8,32 @@ import (
 
 // zstdMaxBufferedEncoders maximum number of not-in-use zstd encoders
 // If the pool of encoders is exhausted then new encoders will be created on the fly
-const zstdMaxBufferedEncoders = 1
+var zstdMaxBufferedEncoders int
```

Comment:
Changing this from a `const` to a `var` feels like a code smell.

Ah, yep. This PR also removes the only use. I propose instead of this change, move `getZstdMaxBufferedEncoders` to here, and renaming it to `zstdMaxBufferedEncoders`.  That way we replace the functionality that the const was providing, and also we can reuse some of the godoc as well.

Reply:
I can do the movement and function rename, but getZstdMaxBufferedEncoders also takes in an argument so it won't be exactly the same as before.

Reply:
Yeah, it won’t be exactly the same, that’s ok.

There’s a kind of common Go style recommendation to avoid `Get` and `get` prefixes. (example: https://google.github.io/styleguide/go/decisions#getters ) The idea is the verb is kind of so generic, that it doesn’t increase the information in the name.